aloe_webdriver/screenshot_failed.py

```diff
@@ -0,0 +1,96 @@
+"""
+Hooks to save screenshots and HTML source of the pages when tests fail.
+
+Assumes a browser instance is stored in ``world.browser``.
+
+Whenever a step fails, the screen shot and the HTML source of the page
+displayed in the browser are saved to the current directory. The file names
+include the feature file name, scenario number and name and, if applicable,
+the example number.
+
+Consider the following feature:
+
+.. code-block:: gherkin
+
+    # features/account.feature
+    Feature: Account management
+
+        Scenario: Log in
+            Given I open the site
+            And I enter username and password
+            And I press "Log in"
+            Then I should see "Logged in"
+
+If there will be no "Logged in" text when expected, screenshot and the page
+source will be saved to, respectively::
+
+    failed_features_account_feature_1_Log_in.png
+    failed_features_account_feature_1_Log_in.html
+"""
+
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+# pylint:disable=redefined-builtin,unused-wildcard-import,wildcard-import
+from builtins import *
+# pylint:enable=redefined-builtin,unused-wildcard-import,wildcard-import
+
+import re
+import os
+
+from aloe import after, before, world
+
+# Pylint cannot infer the attributes on world
+# pylint:disable=no-member
+
+
+FORMAT = 'failed_{feature_file}_{scenario_index}_{scenario_name}{outline_index}'
+
+
+@before.each_example
```

Comment:
I think I used it in the outputter? Worth checking how that works. If it's private API perhaps rename it to `_outline_index` ?